storage/replica_raftstorage.go

```diff
@@ -351,7 +351,10 @@ func (r *Replica) updateRangeInfo() error {
 	// Load the system config.
 	cfg := r.rm.Gossip().GetSystemConfig()
 	if cfg == nil {
-		return fmt.Errorf("no system config available")
+		// This could be before the system config was ever gossiped,
+		// or it expired. Let the gossip callback set the info.
+		log.Warningf("no system config available, cannot determine range MaxBytes")
```

Comment:
I'm not sold; I'd rather this be broken into two functions (or just change this function to take `cfg` as an argument) with differing error semantics.

LGTM as-is for the purpose of fixing the flaky tests.